add neutral tokens to spreadsheet (for 1882)

erperreault opened the pull request:
Issue https://github.com/tobymao/18xx/issues/1332

** Making this generic. 

---

Add spreadsheet functionality to count and display neutral tokens.

## Current behavior (can be misleading, CPR vs others for example):
![before](https://user-images.githubusercontent.com/69309175/105653084-e2e2a700-5e88-11eb-9a28-75ece1cd4f1e.png)

## New behavior:
![after](https://user-images.githubusercontent.com/69309175/105653011-ae6eeb00-5e88-11eb-8e7b-6b7f143567c7.png)

## CN counts neutral tokens as its own:
![CNafter](https://user-images.githubusercontent.com/69309175/105653014-b038ae80-5e88-11eb-9fc3-7e950d1ce958.png)

erperreault commented:
Maybe it would look better as: 

1/1, N

jenf commented:
> Maybe it would look better as:
> 
> 1/1, N

I'd still suggest a separate column as this avoids confusion.

ryandriskel commented:
I'd like to request the column not exist if the game doesn't support neutral tokens.

scottredracecar commented:
I find the +N notation with no extra column to be easy to understand.

erperreault commented:
Set it up as a separate column, here's how it looks:

![column](https://user-images.githubusercontent.com/69309175/106013806-cf9f2980-608a-11eb-9380-9346e42a3613.png)

jenf commented:
Now that I've seen it, I think the previous implementation was probably the better looking. Apologies.

We should add some sort of key to say N=Neutral Token.

erperreault commented:
- Moved back to +N in the Tokens column.
- Added note below tables.

Questions:  Do we need the note, or is the N clear enough? What about "2/2, N" vs "2/2+N"?

Picture showing both:
![note](https://user-images.githubusercontent.com/69309175/106079562-7adbcd00-60e3-11eb-8542-370717fd1f14.png)

ventusignis commented:
At the risk of bikeshedding a possible way to do this is in a fully generic fashion to make a `token_string(entity)` method in game and put the"count + N" logic into 1882 into an implementation of this. Then on top of that make a `def spreadsheet_legend; []; end` method in base and override that in 1882 and use that to build the legend in the view

1822 could later use this fully generic version to display the home, destination, exchange, and spare tokens for the corporations

scottredracecar commented:
Yes, making it universal sounds like a great idea, but implementing this now and then changing it to universal later when it is needed for 1822 also seems reasonable.

jenf commented on lib/engine/game/g_1882.rb:
```diff
@@ -204,6 +204,10 @@ def action_processed(action)
         @sc_company = nil
       end
+
+      def neutral_tokens(corporation)
```
Sorry to keep changing this on you but ventusignis suggestion of a custom method in the game class seemed a good one.

Probably worth having a token_note method that allows a string to be added for 1882 and in future other games.

jenf commented:
Change looks good, remove draft and I'll approve
